### DISCUSSION TITLE
Access data from parent in relation to compare values

### DISCUSSION THREAD

**Discussion by influcenzo** [prisma/prisma#20007] [Q&A] [answered]
I have got two tables (Chat and Message). One Chat can have many Message objects.

A Chat has a column "deletedAt". Representing a timestamp when the chat was last deleted.

Now I want to use findUnique() to get the Chat with all Messages whose the createdAt-date from a Message is greater than the "deletedAt"-date from chat.

```typescript
await this.prismaService.chat.findUnique({
      where: {
        id: chatId,
      },
      include: {
        messages: {
          where: {
            createdAt: {
              gt: chat.deletedAt,
            },
          },
        },
      },
    });
``` 

Is it possible to access the deletedAt property from chat to compare it with the createdAt-date from a Message?

**Comment by nurul3101** [comment, marked as answer]
Hello @influcenzo 👋 

You can compare values of columns from the same table through [fieldReference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#compare-columns-in-the-same-table) feature, but in your case you want to compare columns which are of different tables.

For this case, you would need to split the queries or use a [raw SQL query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

You could:
1. Fetch the chat with its `deletedAt` timestamp.
2. Use the fetched `deletedAt` timestamp to filter messages.

Your query could look something like this:
```ts
  const chat = await this.prismaService.chat.findUnique({
    where: {
      id: chatId,
    },
    select: {
      deletedAt: true
    }
  });

  // Use the fetched deletedAt timestamp to filter messages
  const messages = await this.prismaService.message.findMany({
    where: {
      chatId: chatId,
      createdAt: {
        gt: chat.deletedAt,
      },
    },
  });
```

Hope this helps!

**Reply by influcenzo** [reply]
Thanks for your very quick feedback! I already suspected that it was not possible to build it with prisma in one. 
I am going to solve it via a queryRaw to get the result with a single query.
